File: gems/decomplex/examples/php/state-mesh.php
<?php
class StateMeshExample {
  public function __construct() {
    $this->a = 1;
    $this->b = 2;
  }

  public function writer() {
    $this->a = 3;
  }

  public function reader() {
    return $this->a + $this->b;
  }

  public function a_alias() {
    return $this->a;
  }
}

File: gems/decomplex/examples/c/state-mesh.c
typedef struct StateMeshExample { int a; int b; } StateMeshExample;
void initialize(StateMeshExample *self) { self->a = 1; self->b = 2; }
void writer(StateMeshExample *self) { self->a = 3; }
int reader(StateMeshExample *self) { return self->a + self->b; }
int a_alias(StateMeshExample *self) { return self->a; }
